module/x/gravity/keeper/keeper.go

```diff
@@ -691,21 +691,19 @@ func (k Keeper) MigrateGravityContract(ctx sdk.Context, newBridgeAddress string,
 	store := ctx.KVStore(k.storeKey)
 	store.Set([]byte{types.LatestSignerSetTxNonceKey}, sdk.Uint64ToBigEndian(0))
 
-	// Reset all ethereum event nonces to zero
+	// Reset ethereum event nonce to zero
 	k.setLastObservedEventNonce(ctx, 0)
-	k.iterateEthereumEventVoteRecords(ctx, func(_ []byte, voteRecord *types.EthereumEventVoteRecord) bool {
-		for _, vote := range voteRecord.Votes {
-			val, err := sdk.ValAddressFromBech32(vote)
-
+	// Reset all validators ethereum event nonce to zero
+	delegateKeys := k.getDelegateKeys(ctx)
+	for _, delegateKey := range delegateKeys {
+		if len(delegateKey.ValidatorAddress) != 0 {
```

Comment:
when would there be 0-len addresses?

Reply:
should technically not happen.

After thinking, I decided to remove the safeguard because it the len is nil, then the function will panic allowing us to detect corrupted state during the upgrade